src/makePageElement.ts

```diff
@@ -29,6 +30,12 @@ export async function getPageInfo({
     });
   }
 
+  if (pageData.notFound) {
+    pageObject.page = await get404File({ options });
+    options.res = notFoundResponseEnhancer({ options });
+    pageData.props = (await fetchPageData({ pageObject, options })).props;
```

Comment:
This is actually where all the logic lies.

1. We rewrite the "page" property to be our fallback page (`404` or `_error`).
2. We wire response enhancer hook that will set `statusCode` to 404
3. We refetch props using the new page